pkg/workloadmeta/filter.go

```diff
@@ -0,0 +1,67 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package workloadmeta
+
+// Filter allows a subscriber to filter events by entity kind or event source.
```

Comment:
It may be worth documenting here (or on `NewFilter` that a nil `kinds` or `sources` means "any".  And, generally describing the matching would be useful.